src/eval.c

```diff
@@ -47,6 +47,34 @@ void ldbEnable(client *c);
 void evalGenericCommandWithDebugging(client *c, int evalsha);
 sds ldbCatStackValue(sds s, lua_State *lua, int idx);
 
+typedef struct luaScript {
+    uint64_t flags;
+    robj *body;
+} luaScript;
+
+static void dictLuaScriptDestructor(dict *d, void *val)
+{
+    UNUSED(d);
+    if (val == NULL) return; /* Lazy freeing will set value to NULL. */
+    decrRefCount(((luaScript*)val)->body);
+    zfree(val);
+}
+
+static uint64_t dictStrCaseHash(const void *key) {
+    return dictGenCaseHashFunction((unsigned char*)key, strlen((char*)key));
+}
+
+/* server.lua_scripts sha (as sds string) -> scripts (as robj) cache. */
+dictType shaScriptObjectDictType = {
+        dictStrCaseHash,            /* hash function */
```

Comment:
Changing from `dictSdsCaseHash` to `dictStrCaseHash` feels weird.
Is it due to `dictFind(lctx.lua_scripts, lctx.lua_cur_script)`,?
Maybe we should change it to `sds`?
BTW. Maybe we should remove `lua_cur_script` from `luaCtx`, now `lua_cur_script` is just a temporary variable only used in `evalGenericCommand`.

Reply:
Yes, I changed it because of `dictFind(lctx.lua_scripts, lctx.lua_cur_script)`. I'm not sure what's better to use the original dict definition and use an sds for `lua_cur_script` or to keep `lua_cur_script` as a `char*` and change the dict definition. I wend for the option with better performance.

Re, removing `lua_cur_script` from `luaCtx`, this is a good idea. I'll fix.

Reply:
btw, we can also use dictEntryMetadataBytes for this (will save one malloc, and let us use the normal hash / compare functions)

Reply:
> btw, we can also use dictEntryMetadataBytes...

Decided the added benefit of this is only theoretical and this a relatively small dict. And storing dict values inside the "metadata" field is an abuse of the dict API and will be confusing.

I'm also keeping `dictSdsCaseHash` because I need the custom dict entry anyway and it makes sense to have moved it into _eval.c_.